packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -851,41 +860,51 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
         children: bottomSheets,
         alignment: FractionalOffset.bottomCenter,
       );
-      _addIfNonNull(children, stack, _ScaffoldSlot.bottomSheet);
+      _addChild(children, stack, _ScaffoldSlot.bottomSheet, excludeSemantics: excludeSemantics);
     }
 
-    children.add(new LayoutId(
-      id: _ScaffoldSlot.floatingActionButton,
-      child: new _FloatingActionButtonTransition(
+    _addChild(
+      children,
+      new _FloatingActionButtonTransition(
         child: widget.floatingActionButton,
-      )
-    ));
+      ),
+      _ScaffoldSlot.floatingActionButton,
+      excludeSemantics: excludeSemantics,
+    );
 
     if (themeData.platform == TargetPlatform.iOS) {
-      children.add(new LayoutId(
-        id: _ScaffoldSlot.statusBar,
-        child: new GestureDetector(
+      _addChild(
+        children,
+        new GestureDetector(
           behavior: HitTestBehavior.opaque,
           onTap: _handleStatusBarTap,
-        )
-      ));
+        ),
+        _ScaffoldSlot.statusBar,
+        excludeSemantics: excludeSemantics,
+      );
     }
 
     if (widget.drawer != null) {
       assert(hasDrawer);
-      children.add(new LayoutId(
-        id: _ScaffoldSlot.drawer,
-        child: new DrawerController(
+      _addChild(
+        children,
+        new DrawerController(
           key: _drawerKey,
+          onStateChanged: () {
+            // Rebuild Scaffold when drawer opens/closes to update semantics tree.
```

Comment:
this would be slightly clearer if there was a local member field `_drawerOpen` which you used to determine the exclude semantics logic, and you just set it here inside the setState. it would avoid the empty callback.

if you prefer the current setup, then put this comment inside the setState callback's body.